src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader.cs

```diff
@@ -28,12 +28,17 @@ public BufferReader(in ReadOnlySequence<T> buffer)
             if (buffer.TryGet(ref _nextPosition, out ReadOnlyMemory<T> memory, advance: true))
             {
                 _moreData = true;
-                CurrentSpan = memory.Span;
-                if (CurrentSpan.Length == 0)
+                
+                if (memory.Length == 0)
```

Comment:
@JeremyKuhne, note this change. I wonder if we have other places that need such fixes.